4b embedding quantizer

mikekgfb opened the pull request:
Summary: 4b embedding quantizer

Differential Revision: D56229021

larryliu0820 commented on examples/models/llama2/quantize.py:
```diff
@@ -351,39 +391,53 @@ def quantized_model(self) -> nn.Module:
+                self.weight, self.scales, None, 0, 0, indices, dtype=self.dtype
+            )
+        else:  # 4bit packed
+            return torch.ops.llama_quantized.embedding_4bit.dtype(
```
It seems here it should be `quantized_decomposed::embedding_4bit`

larryliu0820 commented on examples/models/llama2/quantize.py:
```diff
@@ -275,25 +288,41 @@ def replace_embedding_weight_only_grouped_int8_per_channel(
 
-class EmbeddingOnlyInt8QuantHandler:
-    def __init__(self, mod, *, bitwidth: int = 8, group_size: Optional[int] = None):
+class EmbeddingOnlyInt8QuantHandler(QuantHandler):
```
Should we rename this class? Since it's not int8 only anymore.

Commit:
4b embedding quantizer (#3081)

Summary:

4b embedding quantizer

Reviewed By: larryliu0820

Differential Revision: D56229021
